stripped/colabs/Analiza_Korelacji_NASA_stripped.py

```diff
@@ -0,0 +1,122 @@
+
+!pip install pyspark
+
+
+
+import pyspark
+from pyspark.sql import SparkSession
+from pyspark.sql.types import IntegerType, FloatType, StringType, StructType
+spark = SparkSession.builder.appName('Corr_Analysis').getOrCreate()
+
+
+"""
+Teraz zaimportujemy dane
+"""
+
+from google.colab import drive
+import pandas as pd
+drive.mount('/content/drive')
+
+sampled = pd.read_csv('/content/drive/MyDrive/BigMess/NASA/sampled_NASA_200k.csv')
+
+schemat = StructType()
+for i in sampled.columns:
+  if i == "Date":
+    schemat = schemat.add(i, StringType(), True)
+  else:
+    schemat = schemat.add(i, FloatType(), True)
+
+nasa = spark.read.format('csv').option("header", True).schema(schemat).load('/content/drive/MyDrive/BigMess/NASA/NASA.csv')
+nasa.show(5)
+
+
+"""
+Po wczytaniu danych usuniemy kolumny, które nie będą nam potrzebne przy analizie korelacji roślinności z pozostałymi zmiennymi
+"""
+
+# usuwamy z danych kolumny odpowiadające za długość i szerokość geograficzną oraz datę
+columns_to_delete = ['lon', 'lat', 'Date']
+nasa = nasa.drop(*columns_to_delete)
+nasa.show(5)
+
+
+"""
+W naszych rozważaniach zajmiemy się głównie kolumną "GVEG", która odpowiada za wskaźnik "zielonej roślinności". Odnosi się do roślin o zielonych liściach, które są zdolne do fotosyntezy. Poniżej stowrzymy macierz korelacji, a następnie przedstawimy ją w postaci heat mapy.
+"""
+
+from pyspark.ml.stat import Correlation
+from pyspark.ml.feature import VectorAssembler
+
+# convert to vector column first
+vector_col = "corr_features"
+assembler = VectorAssembler(inputCols=nasa.columns, outputCol=vector_col, handleInvalid = 'skip')
+nasa_vector = assembler.transform(nasa).select(vector_col)
+
+# get correlation matrix
+corr_matrix = Correlation.corr(nasa_vector, vector_col)
+
+
+corr_matrix_pd = pd.DataFrame(np.array(corr_matrix.collect()[0][0].toArray()), columns=nasa.columns, index=nasa.columns)
+corr_matrix_np = corr_matrix_pd.to_numpy()
+
+
+# Tworzymy nasza heatmapę. Niestety funkcje pokroju sns.heatmap nie nadają się same w sobie do prezentacji tak dużych macierzy korelacji zatem potrzebne sa im pewne modyfikacje. Poniżej stowrzymy nasz wykres w taki sposób
+# aby po kliknięciu na grafikę ulegała ona przybliżeniu i wyostrzeniu
+
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+print(plt.get_backend())
+
+
+corr_matrix_np[np.triu_indices_from(corr_matrix_np)] = np.nan
+
+fig, ax = plt.subplots(figsize=(24, 18))
+
+hm = sns.heatmap(corr_matrix_np, cbar=True, vmin=-0.5, vmax=0.5,
+                 fmt='.2f', annot_kws={'size': 8}, annot=True,
+                 square=True, cmap=plt.cm.Blues)
+
+ticks = np.arange(corr_matrix_pd.shape[0]) + 0.5
+ax.set_xticks(ticks)
+ax.set_xticklabels(corr_matrix_pd.columns, rotation=90, fontsize=8)
+ax.set_yticks(ticks)
+ax.set_yticklabels(corr_matrix_pd.index, rotation=360, fontsize=8)
+
+ax.set_title('correlation matrix')
+plt.tight_layout()
+plt.savefig("corr_matrix_incl_anno_double.png", dpi=300)
```

Comment:
zapis do pliku, ale go w repo nie widzę. Czemu taka customowa nazwa. I bez ścieżki - ma się zrobić w głównym katalogu? Uporządkujmy to.

Reply:
Gdzie miałby się zapisywać ten obraz w naszym repo?

Reply:
to jest drugorzędny problem. Mnie bardziej martwi, że funkcja zawsze go zapisuje w tym samym miejscu. Jeśli już musi zapisywać (bo rozumiem, że to kwestia że heatmapy są spore i trudno je przejrzeć z notatnika) to miejsce zapisu brzmi jak oczywisty parametr.

Reply:
Generalnie rzecz biorąc w notatniku colabowym działa to tak, że ten obraz zapisuje się w środowisku, na którym odpalamy colaba. Przeszukałem stackoverflow i nie udało mi się znaleźć opcji zapisu obrazu z notatnika colabowego do konkretnej ścieżki na dysku lokalnym.  Zamiast tego istnieje opcja pobrania go którą z resztą dodałem. Niemniej jednak wówczas obraz pobiera się do folderu "pobrane" i tutaj też nie udało mi się znaleźć opcji pobrania do konkretnego miejsca na dysku lokalnym na naszym komputerze.

Reply:
zrobione